src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -418,7 +418,11 @@ RouterModifyTaskForShardInterval(Query *originalQuery, ShardInterval *shardInter
 	DistTableCacheEntry *cacheEntry = DistributedTableCacheEntry(distributedTableId);
 
 	RelationRestrictionContext *copiedRestrictionContext =
-		CopyRelationRestrictionContext(restrictionContext);
+		CopyRelationRestrictionContext(
+			plannerRestrictionContext->relationRestrictionContext);
+	RelationRestrictionContext *originalRestrictionContext =
```

Comment:
Why do we change `RouterModifyTaskForShardInterval ()` at all? I cannot see this being used? Can't we only change the parameters when this function is called?